src/goto-instrument/contracts/assigns.cpp

```diff
@@ -93,6 +207,17 @@ assigns_clauset::conditional_address_ranget::generate_snapshot_instructions()
   source_locationt location_no_checks = location;
   disable_pointer_checks(location_no_checks);
 
+  null_message_handlert null_handler;
+
+  // clean up side effects from the guard expression
+  // we want checks on the guard evaluation instructions because it is user code
+  cleanert cleaner(parent.symbol_table, null_handler);
+  exprt clean_guard(guarded_slice.guard);
+  const auto &mode = parent.symbol_table.lookup_ref(parent.function_name).mode;
+  cleaner.clean(clean_guard, instructions, mode);
+  for(auto &inst : instructions.instructions)
+    inst.source_location_nonconst() = location;
```

Comment:
Adding that `location` to `clean_guard` should suffice for source locations to be set up correctly, no need for this `for` loop once that's done.